FieldOpt/Hdf5SummaryReader/hdf5_summary_reader.cpp

```diff
@@ -8,27 +8,176 @@ Hdf5SummaryReader::Hdf5SummaryReader(const std::string file_path)
   DATASET_NAME_TIMES("TIMES"),
   GROUP_NAME_FLOW_TRANSPORT("FLOW_TRANSPORT"),
   DATASET_NAME_ACTIVE_CELLS("ACTIVE_CELLS"),
-  DATASET_NAME_WELL_STATES("WELL_STATES")
+  DATASET_NAME_WELL_STATES("WELL_STATES"),
+  DATASET_NAME_PRESSURE("PTZ"),
+  DATASET_NAME_SATURATION("GRIDPROPTIME")
 {
     readTimeVector(file_path);
-    readActiveCells(file_path);
     readWellStates(file_path);
+    readActiveCells(file_path);
```

Comment:
I don't think the cell states should not be read in the constructor for the summary reader. This is because normally, at least the way I see it, FieldOpt has no use for it. 

I suggest to either add a boolean flag that defaults to false (e.g. `bool read_cell_data=false`), or separate this out into a new method `read_cell_data()` that can be called after reading the other parts of the summary. This is because this can, especially for large reservoirs, be very memory and time consuming,